tests/unit/test_annotate.py

```diff
@@ -806,3 +741,85 @@ def test_delete_variable(sample_netcdf4_file_test05, group_name):
         delete_variable(test_datatree, full_path)
         assert variable_name not in test_datatree[group_name].ds.data_vars
         assert len(test_datatree[group_name].ds.data_vars) == initial_var_count - 1
+
+
+@pytest.mark.parametrize(
+    'ancestor_path, dim_name, expected',
+    [('/', 'x', '/x'), ('/sub_group', 'x', '/sub_group/x')],
+)
+def test_construct_dim_path(ancestor_path, dim_name, expected):
+    """Verify the correct full dimension path is constructed."""
+    assert construct_dim_path(ancestor_path, dim_name) == expected
+
+
+@pytest.mark.parametrize(
+    'variable_path, expected', [('/variable_one', True), ('/variable_two', False)]
+)
+def test_has_dimension_override(sample_varinfo_test06, variable_path, expected):
+    """Ensure variables correctly evaluated to contain non-null dimensions override."""
+    assert has_dimension_override(sample_varinfo_test06, variable_path) == expected
+
+
+@pytest.mark.parametrize(
+    'variables_to_create, expected',
+    [
+        (
+            {'/x_root', '/y_root', '/sub_group/x', '/sub_group/y'},
+            {'/x_root', '/y_root', '/sub_group/x', '/sub_group/y'},
+        ),
+        ({'/x_root', '/y_root'}, {'/x_root', '/y_root'}),
+        ({'/sub_group/x', '/sub_group/y'}, {'/sub_group/x', '/sub_group/y'}),
+        (set(), set()),
+    ],
+)
+def test_get_new_dimension_variables(
+    sample_netcdf4_file_test07, variables_to_create, expected
+):
+    """Ensure the correct dimensions are returned.
+
+    This test ensures that the dimension variables present in both the datatree and the
+    `variables_to_create` set are returned.
+    """
+    with xr.open_datatree(sample_netcdf4_file_test07) as test_datatree:
+        assert (
+            get_new_dimension_variables(test_datatree, variables_to_create) == expected
+        )
```

Comment:
I don't really understand this test. 

It might help if there was a test that asked for more variables to create than exist on the file. because that passes too. 

I'm not even sure if that' useful information or if that would ever happen 
e.g.

``` python
        (
            {'/y_root',  '/sub_group/y', '/missing/dimension/z'},
            {'/y_root',  '/sub_group/y'},
        ),
```